e2e/tests/node.go

```diff
@@ -27,6 +27,9 @@ var (
 	// defaultPathURL used by curl commands to send relay requests
 	defaultPathURL = os.Getenv("PATH_URL")
 	// defaultPathHostOverride overrides the host in the URL used to send requests
+	// Since the current DevNet infrastructure does not support arbitrary subdomains,
+	// this is used to specify the host to connect to and the full host (with the service as a subdomain)
+	// will be sent in the "Host" request header.
```

Comment:
Note that comments like `Since the current...` are not "evergreen".

This makes sense to me and you today because we have context and are reading it in the scope of this PR. Will it make sense to a random person on the internet in 12 months? Will it even be relevant then?

Ways to improve this:
 - Reference the PR where this is added
 - Add a date when this i being added
 - Add a TODO for future readers
 - Etc...